### ISSUE TITLE
Links in docstring summary break when the summary is included on another page

### ISSUE THREAD

**Issue by not-my-profile** [twisted/pydoctor#478]
```python
"""
Provides L{foo}.
"""

def foo():
    pass
```

![image](https://user-images.githubusercontent.com/73739153/152488286-ff3007a6-0f0e-4574-86a3-73584079efe4.png)

On that page `foo` links to `#foo` but it should link to `bar.html#foo`.

It's worth noting that ReST links break even worse:

```python
"""
Implements `Foo`_

.. _Foo: http://example.com
"""
```

results in

![image](https://user-images.githubusercontent.com/73739153/152488604-c24b3032-7279-4c95-959d-b4d14e2868ef.png)

with `` `Foo`_ `` linking to `#id1` for some reason.

**Comment by tristanlatr** [comment]
this is because we are generating the summary from the source code and not from the parsed docstring. 

think this will be fixed when fixing #86

**Comment by tristanlatr** [comment]
What do you mean exactly by "is included on another page"? 

I think the behaviour your describing is related to this code: https://github.com/twisted/pydoctor/blob/master/pydoctor/epydoc2stan.py#L63-L67

Because we are passing the source here: https://github.com/twisted/pydoctor/blob/5ed88ac47a781b96a64adbe1be3fc801d98feb77/pydoctor/epydoc2stan.py#L788

This is an optimization not to reload the page when browsing the docs. I'm not sure if we should change this behaviour, maybe as an option?

**Comment by not-my-profile** [comment]
Ah yes that code indeed causes the broken link in my Epydoc example. For ReST the problem is as you said that the docstring is split before parsing and not after. I think to support ReST anchors would need even need more processing to support e.g:


  ```rest
  This module does `foo`_.

  Some text.

  .. _foo:
  Foo
  ---
  Description of foo.
  ```

**Comment by tristanlatr** [comment]
For the pages to be safely included in other pages, we could introduce option --absolute-urls, would it solve your issue ?

**Comment by not-my-profile** [comment]
No we should properly fix these bugs, we don't need absolute URLs for that, which would harm the user experience.

**Comment by tristanlatr** [comment]
The links, even fixed in the summary pages, will still be relative. So if relative links doesn’t work for you, it won’t work. This is what I meant.

**Comment by not-my-profile** [comment]
Relative links work great ... we just have to use the right ones on the right page.

**Comment by tristanlatr** [comment]
Ok sorry. There was a misunderstanding. 

The issue resides in the summary that are generated for the module/class/names indexes pages. Links starting by “#” are generated where it should use the whole file name.

We need to format summary with the same source to resolve links correctly, but the behavior of taglink() function is annoying for this case, so we should be able to disable this behavior with an argument. We could introduce a new argument for the _EpydocLinker, something like `_EpydocLinker(ob, same_page_optimization=False)`, then we pass that option around to the summary pages generation code.

**Comment by tristanlatr** [comment]
Re-opening this issue since #498 fixed only the first part of the issue. 

The other issue will be fixed while fixing #86.
